File: RosTopic/src/package/src/publisher_node.cpp
#include "ros/ros.h"
#include "package/topic.h"

int main(int argc, char **argv){
    ros::init(argc,argv,"publisher_node_cpp");

    ros::NodeHandle nh;

    ros::Publisher data_publisher = nh.advertise<package::topic>("cpp_topic", 1000);

    ros::Rate loop_rate(1); //1hz

    while(ros::ok()){
        package::topic send;

        send.usia = 18;
        send.tinggi = 1.68;
        send.nama = "Brian Ricardo Tamin";
        send.kehadiran = true;

        ROS_INFO("nama: %s", send.nama.c_str());
        ROS_INFO("usia: %ld", send.usia);
        ROS_INFO("tinggi: %f", send.tinggi);
        ROS_INFO("kehadiran: %s\n", send.kehadiran ? "hadir" : "tidak hadir");

        data_publisher.publish(send);

        ros::spinOnce();
        loop_rate.sleep();
    }


    return 0 ;
}

File: TugasModul4/RosTopic/src/package/src/publisher_node.py
#!/usr/bin/env python

import rospy
from package.msg import topic

def publisher_node_py():
    rospy.init_node('publisher_node_py', anonymous=True)

    pub_data = rospy.Publisher('py_topic', topic, queue_size=1000)

    loop_rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        send = topic()
        send.usia = 18
        send.tinggi = 1.68
        send.nama = "Brian Ricardo Tamin"
        send.kehadiran = True

        rospy.loginfo("nama: %s", send.nama)
        rospy.loginfo("usia: %d", send.usia)
        rospy.loginfo("tinggi: %f", send.tinggi)
        rospy.loginfo("kehadiran: %s\n", "hadir" if send.kehadiran else "tidak hadir")

        pub_data.publish(send)

        loop_rate.sleep()

if __name__ == '__main__':
    try:
        publisher_node_py()
    except rospy.ROSInterruptException:
        pass
